src/utils/first.ts

```diff
@@ -1,12 +1,12 @@
 // Return the first non-null or -undefined result from an array of
 // maybe-sync, maybe-promise-returning functions
-export default function first<T>(
+export default function first<T, Context>(
```

Comment:
Not 100% happy about this as this new context handling is not too obvious from the function name. What about making the context and explicit parameter of this function and *not* add the context if we do not pass the parameter? In that case, I would be ok with keeping the function name as it is.

That would make the comment above true again. Otherwise, this function is now very specific to certain usages and may have unintended side-effects.

Reply:
Also, why did you perform this change? It touches quite a bit of existing functionality.

Reply:
This function is only used for calling plugin hooks. And all plugin hooks now need context to be passed through it. It seems like a natural extension to me that the caller should set the context for the individual functions, as if the caller calls all functions as a single function.